### PR TITLE
KFSPTS-29347: Upgrade to the 04/19/2023 version of financials.

### PR DESCRIPTION
**Pull request by nkimble**

### CODE REVIEW

**Comment by jhulslander** [comment]
retest this please

**Comment by jhulslander** [comment]
retest this please

File: src/main/resources/edu/cornell/kfs/module/purap/cu-ojb-purap.xml [hunk 1]
```diff
@@ -3309,162 +3333,115 @@
...
+                               collection-class="org.apache.ojb.broker.util.collections.ManageableArrayList"
+                               auto-retrieve="true" auto-update="none" auto-delete="none">
+            <orderby name="transactionLedgerEntrySequenceNumber" sort="ASC"/>
+            <query-customizer class="org.kuali.kfs.sys.dataaccess.impl.OjbQueryCustomizer">
+                <attribute attribute-name="documentNumber" attribute-value="parent.documentNumber"/>
+            </query-customizer>
+        </collection-descriptor>
+        
+        <!-- Cornell Customization -->
+        <reference-descriptor name="wireTransfer" class-ref="org.kuali.kfs.sys.businessobject.PaymentSourceWireTransfer"
```
Comment (db483) [inline src/main/resources/edu/cornell/kfs/module/purap/cu-ojb-purap.xml:3440]: should this be moved up with the other reference-descriptor entries?

**Review by db483 — CHANGES_REQUESTED** [review]
I am still reviewing the upgrade but I added some notes and some things that might need further attention. Thank you.

File: pom.xml [hunk 2]
```diff
@@ -1103,6 +1103,7 @@
             <groupId>com.google.code.gson</groupId>
             <artifactId>gson</artifactId>
             <version>${gson.version}</version>
+            <scope>compile</scope>
```
Comment (db483) [inline pom.xml:1106]: Should this be 'runtime'?

Comment (db483) [reply pom.xml:1106]: updated

File: src/main/java/edu/cornell/kfs/module/purap/document/authorization/CuPaymentRequestDocumentPresentationController.java [hunk 3]
```diff
@@ -42,7 +40,6 @@ public Set<String> getEditModes(final Document document) {
             editModes.add(CUPaymentRequestEditMode.WAIVE_WIRE_FEE_EDITABLE);
             // KFSPTS-1891
             editModes.add(KfsAuthorizationConstants.TransactionalEditMode.FRN_ENTRY);
-            editModes.add(KfsAuthorizationConstants.TransactionalEditMode.WIRE_ENTRY);
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/edu/cornell/kfs/module/purap/document/authorization/CuPaymentRequestDocumentPresentationController.java:45]: This method call super and base code is adding WIRE_ENTRY when doc in PaymentMethodReview node so this changes should be fine.

File: src/main/java/edu/cornell/kfs/module/purap/service/impl/CuElectronicInvoiceHelperServiceImpl.java [hunk 4]
```diff
@@ -766,7 +766,7 @@ protected PaymentRequestDocument initializePaymentRequestDocument(final Electron
         //KFSPTS-1891
         boolean hasPaymentMethodCode = false;
         if ( preqDoc instanceof PaymentRequestDocument ) {
-            String vendorPaymentMethodCode = ((VendorDetailExtension)poDoc.getVendorDetail().getExtension()).getDefaultB2BPaymentMethodCode();
+            String vendorPaymentMethodCode = ((VendorDetail)poDoc.getVendorDetail()).getDefaultPaymentMethodCode();
             if ( StringUtils.isNotEmpty(vendorPaymentMethodCode) ) { 
                 ((CuPaymentRequestDocument)preqDoc).setPaymentMethodCode(vendorPaymentMethodCode);
                 hasPaymentMethodCode = true;
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/edu/cornell/kfs/module/purap/service/impl/CuElectronicInvoiceHelperServiceImpl.java:772]: updated

File: src/main/java/edu/cornell/kfs/module/purap/service/impl/CuPdpExtractService.java [hunk 5]
```diff
@@ -14,15 +14,12 @@
...
 import org.kuali.kfs.kew.api.document.attribute.DocumentAttributeIndexingQueue;
 import org.kuali.kfs.kim.impl.identity.Person;
 
 import edu.cornell.kfs.module.purap.CUPurapConstants;
 
 public class CuPdpExtractService extends PdpExtractService {
-    private DocumentTypeService documentTypeService;
     
     @Override
     protected void updatePaymentRequest(
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/edu/cornell/kfs/module/purap/service/impl/CuPdpExtractService.java:25]: updated

File: src/main/resources/edu/cornell/kfs/sys/businessobject/datadictionary/PaymentMethod.xml [hunk 6]
```diff
@@ -20,23 +20,20 @@
...
-    <bean id="PaymentMethod-extension-displayOnVendorDocument" parent="PaymentMethod-extension-displayOnVendorDocument-parentBean"/>
-    <bean id="PaymentMethod-extension-displayOnVendorDocument-parentBean" abstract="true" parent="PaymentMethodExtendedAttribute-displayOnVendorDocument"
-            p:name="extension.displayOnVendorDocument"/>
-            
     <bean id="PaymentMethod-extension-displayOnRecurringDVDocument" parent="PaymentMethod-extension-displayOnRecurringDVDocument-parentBean"/>
     <bean id="PaymentMethod-extension-displayOnRecurringDVDocument-parentBean" abstract="true" parent="PaymentMethodExtendedAttribute-displayOnRecurringDVDocument"
             p:name="extension.displayOnRecurringDVDocument"/>
             
-        <bean id="PaymentMethod-lookupDefinition" parent="PaymentMethod-lookupDefinition-parentBean" >
+    <bean id="PaymentMethod-extension-additionalCreditMemoDataCode" parent="PaymentMethod-extension-additionalCreditMemoDataCode-parentBean"/>
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/resources/edu/cornell/kfs/sys/businessobject/datadictionary/PaymentMethod.xml:27]: additionalCreditMemoDataCode needs to be added to the inquiry and maintenance screens

[the remark this answers is not in the document]
Comment (db483) [reply src/main/resources/edu/cornell/kfs/sys/businessobject/datadictionary/PaymentMethod.xml:27]: updated

File: pom.xml (hunk 2 above)
Comment (db483) [reply pom.xml:1106]: needed at compile time

File: src/main/resources/edu/cornell/kfs/module/purap/cu-ojb-purap.xml (hunk 1 above)
Comment (db483) [reply src/main/resources/edu/cornell/kfs/module/purap/cu-ojb-purap.xml:3440]: moved

**Pull request closed without merging**